src/System.Net.Http/src/System/Net/HttpClient.cs

```diff
@@ -156,41 +156,124 @@ public Task<Stream> GetStreamAsync(string requestUri)
         // Unbuffered by default
         public Task<Stream> GetStreamAsync(Uri requestUri)
         {
-            return GetContentAsync(requestUri, HttpCompletionOption.ResponseHeadersRead, Stream.Null,
+            return GetStreamContentAsync(requestUri, HttpCompletionOption.ResponseHeadersRead,
                 content => content.ReadAsStreamAsync());
         }
 
-        private Task<T> GetContentAsync<T>(Uri requestUri, HttpCompletionOption completionOption, T defaultValue,
-            Func<HttpContent, Task<T>> readAs)
+        private Task<string> GetStringContentAsync(Uri requestUri, HttpCompletionOption completionOption, 
+            Func<HttpContent, Task<string>> readAs)
         {
-            TaskCompletionSource<T> tcs = new TaskCompletionSource<T>();
+            var tcs = new TaskCompletionSource<string>(readAs);
```

Comment:
This ```readAs``` delegate is now going to be exposed from the returned ```Task```'s AsyncState property.  That's probably undesirable, but it's up to @davidsh et al whether they're comfortable with that.